src/NUnitFramework/tests/Assertions/AssertThrowsAsyncTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.TestUtilities;
+
+namespace NUnit.Framework.Tests.Assertions
+{
+    [TestFixture]
+    public class AssertThrowsAsyncTests
+    {
```

Comment:
The tests in this class are equivalent to the ones in `AssertThrowsTests`, but with the async version. Some of them have two variants:
- `*Sync`: to test the case where the test delegate fails synchronously (i.e. throws before yielding control back to the caller)
- `*Async`: to test the case where the test delegate fails asynchronously (i.e. sucessfully returns a task that will fail later)

Since `ThrowsAsync` isn't using async/await, it has to handle those cases separately, so there are separate tests to cover both cases.